tests/unit/introspection-2.tcl

```diff
@@ -101,13 +101,60 @@ start_server {tags {"introspection"}} {
         assert_equal {key1 key2} [r command getkeys lcs key1 key2]
     }
 
+    test "COMMAND LIST syntax error" {
+        assert_error "ERR syntax error*" {r command list bad_arg}
+        assert_error "ERR syntax error*" {r command list filterby bad_arg}
+        assert_error "ERR syntax error*" {r command list filterby bad_arg bad_arg2}
+    }
+
+    test "COMMAND LIST WITHOUT FILTERBY" {
+        set reply [r command list]
+        assert_match "*set*" $reply
+        assert_match "*client|list*" $reply
+    }
+
+    test "COMMAND LIST FILTERBY ACLCAT against non existing category" {
+        assert_equal {} [r command list filterby aclcat non_existing_category]
+    }
+
     test "COMMAND LIST FILTERBY ACLCAT" {
         set reply [r command list filterby aclcat hyperloglog]
-        assert_equal [lsort $reply] {pfadd pfcount pfdebug pfmerge pfselftest}
+        assert_match "*pfadd*" $reply
+        assert_match "*pfcount*" $reply
+    }
+
+    test "COMMAND LIST FILTERBY ACLCAT - SUBCOMMANDS" {
+        set reply [r command list filterby aclcat scripting]
+        assert_match "*eval*" $reply
+        assert_match "*script|kill*" $reply
+        assert_match "*function|list*" $reply
     }
 
     test "COMMAND LIST FILTERBY PATTERN" {
         set reply [r command list filterby pattern pf*]
-        assert_equal [lsort $reply] {pfadd pfcount pfdebug pfmerge pfselftest}
+        assert_match "*pfadd*" $reply
+        assert_match "*pfcount*" $reply
+    }
+
+    test "COMMAND LIST FILTERBY PATTERN - SUBCOMMANDS" {
+        # Return the parent command and all the subcommands below it.
+        set reply [r command list filterby pattern config*]
+        assert_match "*config*" $reply
+        assert_match "*config|get*" $reply
```

Comment:
if we would have emitted only the second one, the first one would still be hit.
let's convert this to a list and do a lookup into it.